### DISCUSSION TITLE
forge/cast option to execute transactions in browser

### DISCUSSION THREAD

**Discussion by frolic** [foundry-rs/foundry#3682] [Ideas]
Juggling private keys scares me a little. If I've got all my keys/balances held within Rainbow, MetaMask, etc., it'd be great if there was an option when running `forge`/`cast` to pop open a browser window, allow me to connect with one of those wallets, and then approve transactions being executed by `forge`/`cast`. I assume this would operate similar to the hardware wallet options, but haven't used those.

If this is a desirable feature, any way I can help with this? Happy to take on some of the frontend work. I assume we'll need:
- frontend that allows you connect with your preferred wallet (via RainbowKit or ConnectKit)
- communication bridge between frontend and CLI to know when you're connected and to forward transactions/calls

**Comment by mds1** [comment]
Related to https://github.com/foundry-rs/foundry/issues/67

I do agree this would be a nice feature, will defer to @gakonst on how you can best help implement 🙂

**Reply by sambacha** [reply]
you can download and run this, its a fork of truffle's dashboard, https://github.com/manifoldfinance/walletbus this works with cast, but not forge last I checked. 

you can send tx requests from shell to your browser using this
